server/middlewares/auth/userAuth.ts

```diff
@@ -0,0 +1,20 @@
+import { Request, Response, NextFunction } from 'express';
+
+import { VerfiyAuth } from '../../utils/auth';
+
+const userAuth = async (req: Request, res: Response, next: NextFunction) => {
+  const { JWT_SECRET } = req.cookies;
+
+  if (!JWT_SECRET) {
+    res.status(401).json({ message: 'Login First!' });
+  } else {
+    try {
+      await VerfiyAuth(JWT_SECRET);
+      next();
```

Comment:
You received data in checkAuth route from this middleware, so you need to send it here.
Also don't use try and catch and handle it in the server middleware